### PR TITLE
removed cache and using filter

### PR DESCRIPTION
**Pull request by patduin**
reimplementation of the issue raised here: https://github.com/ExpediaGroup/waggle-dance/pull/295
@flaming-archer  please have a look :)

### CODE REVIEW

**Comment by flaming-archer** [comment]
LGTM,  testing in my environment is fine.

**Comment by patduin** [comment]
> LGTM, testing in my environment is fine.

Thanks @flaming-archer

**Comment by flaming-archer** [comment]
I think we can merge this branch now, haha.

**Comment by patduin** [comment]
> I think we can merge this branch now, haha.

please approve, I've also ask someone else to have another approval, waiting for that.

**Comment by flaming-archer** [comment]
> > I think we can merge this branch now, haha.
> 
> please approve, I've also ask someone else to have another approval, waiting for that.

You didn't ask me to approve. Yes , I approve.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 ### Fixed
 * Added lombok
 * Fixed test cases
+* Fixed issue where the primare metastore was not applying the allow filter to validate database clashes from other metastores.
```
Comment (abhimanyugupta07) [inline CHANGELOG.md:20]:
```suggestion
* Fixed issue where the primary metastore was not applying the allow filter to validate database clashes from other metastores.
```

**Review by abhimanyugupta07 — APPROVED** [review]

**Pull request merged**